src/pages/index.js

```diff
@@ -0,0 +1,300 @@
+
+// Delete form elements
+const deleteModal = document.querySelector("#delete-modal");
+
+// Card related elements
+const cardTemplate = document.querySelector("#card-template");
+const cardsList = document.querySelector(".cards__list");
+
+function getCardElement(data, userInfo) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+
+  const cardNameEl = cardElement.querySelector(".card__title");
+  const cardImageEl = cardElement.querySelector(".card__image");
+  const cardLikeButton = cardElement.querySelector(".card__like-button");
+  const cardDeleteButton = cardElement.querySelector(".card__delete-button");
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  // Ensure data.likes is an array
+  const likes = data.likes || [];
+
+  // Set initial like state based on whether the user has liked the card
+  if (likes.some((like) => like._id === userInfo._id)) {
+    cardLikeButton.classList.add("card__like-button_liked");
+  }
+
+  // Like/Dislike logic
+  cardLikeButton.addEventListener("click", () => {
+    const isLiked = cardLikeButton.classList.contains(
+      "card__like-button_liked"
+    );
+
+    if (isLiked) {
+      api
+        .dislikeCard(data._id)
+        .then(() => {
+          cardLikeButton.classList.remove("card__like-button_liked");
+        })
+        .catch((err) => {
+          console.error("Error disliking card:", err);
+        });
+    } else {
+      api
+        .likeCard(data._id)
+        .then(() => {
+          cardLikeButton.classList.add("card__like-button_liked");
+        })
+        .catch((err) => {
+          console.error("Error liking card:", err);
+        });
+    }
+  });
+
+  // Delete card logic
+  cardDeleteButton.addEventListener("click", () => {
+    handleDeleteCardSubmit(cardElement, data._id);
+  });
+
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalImageElement.src = data.link;
+    previewModalCaptionElement.textContent = data.name;
+    previewModalImageElement.alt = data.name;
+  });
+
+  return cardElement;
+}
+
+function closeOverlay(evt) {
+  if (evt.target.classList.contains("modal")) {
+    closeModal(evt.target);
+  }
+}
+
+function handleEsc(evt) {
+  if (evt.key === "Escape") {
+    const openModal = document.querySelector(".modal_opened");
+    if (openModal) {
+      closeModal(openModal);
+    }
+  }
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  modal.addEventListener("click", closeOverlay);
+  document.addEventListener("keydown", handleEsc);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  modal.removeEventListener("click", closeOverlay);
+  document.removeEventListener("keydown", handleEsc);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, "Saving...");
```

Comment:
`"Saving..."` is not needed as the 2nd argument because it should be `true` when you want to show `Saving...`


```js
setButtonText(submitBtn, true);

```